lib/ProviderAws.js

```diff
@@ -175,26 +180,30 @@ module.exports = function(S) {
 
       stage = stage ? stage.toUpperCase() : null;
 
-      // implicitly already in the config...
-
-      this.addConfigurationCredentials(credentials, S.config);                      // use the given configuration credentials if they are the only available credentials.
-      // first from environment
-      this.addEnvironmentCredentials(credentials, 'AWS');                                 // allow for Amazon standard credential environment variable prefix.
-      this.addEnvironmentCredentials(credentials, 'SERVERLESS_ADMIN_AWS');                // but override with more specific credentials if these are also provided.
-      this.addEnvironmentCredentials(credentials, 'AWS_' + stage);                        // and also override these with the Amazon standard *stage specific* credential environment variable prefix.
-      this.addEnvironmentCredentials(credentials, 'SERVERLESS_ADMIN_AWS_' + stage);       // finally override all prior with Serverless prefixed *stage specific* credentials if these are also provided.
-      // next from profile
-      this.addProfileCredentials(credentials, 'AWS');                                     // allow for generic Amazon standard prefix based profile declaration
-      this.addProfileCredentials(credentials, 'SERVERLESS_ADMIN_AWS');                    // allow for generic Serverless standard prefix based profile declaration
-      this.addProfileCredentials(credentials, 'AWS_' + stage);                            // allow for *stage specific* Amazon standard prefix based profile declaration
-      this.addProfileCredentials(credentials, 'SERVERLESS_ADMIN_AWS_' + stage);           // allow for *stage specific* Serverless standard prefix based profile declaration
-      // if they aren't loaded now, the credentials weren't provided by a valid means
-
-      if (!credentials.accessKeyId || !credentials.secretAccessKey) {
-        throw new SError('Cant find AWS credentials', SError.errorCodes.MISSING_AWS_CREDS);
-      }
+      return BbPromise.resolve(credentials)
+
+        // implicitly already in the config...
+        .then(() => this.addConfigurationCredentials(credentials, S.config))                      // use the given configuration credentials if they are the only available credentials.
```

Comment:
Neither addConfigurationCredentials nor addEnvironmentCredentials are asynchronous so that wrapping them in this then syntax seems a bit odd.

Instead, line 183 can be moved to just before 195 below.